src/logic/safe/store/actions/utils.ts

```diff
@@ -21,14 +21,12 @@ export const getNewTxNonce = async (
   lastTx: TxServiceModel | null,
   safeInstance: GnosisSafe,
 ): Promise<string> => {
-  if (typeof txNonce === 'string' && !Number.isInteger(Number.parseInt(txNonce, 10))) {
-    return lastTx === null
-      ? // use current's safe nonce as fallback
-        (await safeInstance.methods.nonce().call()).toString()
-      : `${lastTx.nonce + 1}`
+  if (txNonce) {
```

Comment:
are you sure about this change? shouldn't it be txNonce + 1?

Reply:
Also, there's not a single test where this value is defined which makes me question whether we need it at all and what it was used for

Reply:
Ok, seems that it is used when we have already pre-calculated  the nonce. Would be good to have a test for this

Reply:
done